wolfcrypt/src/random.c

```diff
@@ -2741,14 +2745,26 @@ int wc_GenerateSeed(OS_Seed* os, byte* output, word32 sz)
         }
     #endif /* HAVE_INTEL_RDSEED */
 
+#ifdef WIN_REUSE_CRYPT_HANDLE
+    if (gHandle == 0) {
```

Comment:
is `wc_GenerateSeed` supposed to be thread safe? if it is, then this check is not safe.

Reply:
Hi @redbaron , no this is not thread-safe and it will need fixed before this can be considered. It was already pointed out by a few other wolfSSL engineers.